templates/includes/pagination.html

```diff
@@ -0,0 +1,20 @@
+{% if is_paginated %}
+  <ul class="pagination">
+    {% if page_obj.has_previous %}
+      <li class="page-item">
+        <a href="?page= {{ page_obj.previous_page_number }}" class="page-link">prev</a>
```

Comment:
There is an extra space after the '=' in the href attribute. It should be '?page={{ page_obj.previous_page_number }}' without the space.